src/game/eop.js

```diff
@@ -116,321 +109,324 @@ export const ElevationOfPrivilege = Game({
-        for (let i=0; i<scores.length; i++) {
-          if (scores[i] > max) {
-            winner = i;
-            max = scores[i];
-          }
-        }
-        return winner;
-      }
-    },
-    phases: {
-      threats: {
-        turnOrder: {
-          first: function first(G, ctx) {
-            return ctx.playOrderPos;
-          },
-          next: function next(G, ctx) {
-            return ctx.playOrderPos;
-          },
-          actionPlayers: {
-            value: (G, ctx) => getPlayers(ctx.numPlayers), // hack, ANY wasn't working because of int actionPlayers
-          }
-        },
-        next: 'play',
-        allowedMoves: ['addOrUpdateThreat', 'deleteThreat', 'pass', 'selectDiagram', 'selectComponent', 'selectThreat', 'toggleModal', 'toggleModalUpdate', 'updateThreat'],
-        endPhaseIf: (G, ctx) => {
-          let passed = [...G.passed];
-          return passed.length >= ctx.numPlayers;
-        },
-        onPhaseEnd: (G, ctx) => {
-          let dealt = [...G.dealt];
-          let order = [...G.order];
-          let suit = G.suit;
-          let dealtBy = G.dealtBy;
-          let scores = [...G.scores];
-          let round = G.round;
-          let lastWinner = G.lastWinner;
-
-          // calculate the scores
-          if (dealt.length >= ctx.numPlayers) {
-            let idx = getWinner(suit, dealt);
-            lastWinner = order[idx];
-    
-            scores[lastWinner]++;
-    
-            order = [];
-            for (let i=0; i<ctx.numPlayers; i++) {
-              order.push(i);
-            }
-            for (let i=0; i<lastWinner; i++) {
-              let element = order.shift();
-              order.push(element);
-            }
-    
-            dealt = [];
-            suit = "";
-            dealtBy = "";
-            round++;
-          }
-
-          return {
-            ...G,
-            dealt,
-            lastWinner,
-            dealtBy,
-            order,
-            suit,
-            scores,
-            round,
-            passed: [], // reset the passed array when the phase ends
-          }
-        },
-      },
-      play: {
-        turnOrder: {
-          playOrder: function playOrder(G) {
-            return G.order;
-          },
-          first: function first(G, ctx) {
-            return G.playOrderPos % ctx.playOrder.length;
-          },
-          next: function next(G, ctx) {
-            return (ctx.playOrderPos + 1) % ctx.playOrder.length;
-          }
-        },
-        next: 'threats',
-        allowedMoves: ['draw', 'selectDiagram', 'selectComponent', 'selectThreat'],
-        onTurnEnd: (G, ctx) => {
-          ctx.events.endPhase();
-          return {
-            ...G,
-            playOrderPos: G.playOrderPos + 1,
-          }
-        },
-      },
+  // only update score if it's a new threat
+  if (G.threat.new) {
+    scores[ctx.playerID]++;
+  }
+
+  // TODO: have a cleaner or readable approach to updating this object
```

Comment:
Do it!  :-)

Reply:
I'm not really sure of a cleaner way to do this, I might have a think about it and come back to it.